pyscf_ipu/nanoDFT/nanoDFT.py

```diff
@@ -66,10 +66,14 @@ def nanoDFT_iteration(i, vals, opts, mol):
     if opts.diis: H, diis_history = DIIS(i, H, density_matrix, O, diis_history, opts)           # H_{i+1}=c_1H_i+...+c9H_{i-9}.
 
     # Step 2: Solve eigh (L_inv turns generalized eigh into eigh).
-    eigvects = L_inv.T @ linalg_eigh(L_inv @ H @ L_inv.T, opts)[1]                              # (N, N)
+    linalg_input = L_inv @ H @ L_inv.T
```

Comment:
We might consider merging all the eigh tricks together:
```
L_inv.T @ prev_eigvects @ eigh(prev_eigvects.T @ L_inv @ H @ L_inv.T @ prev_eigvects)
```
This hints at the trick to turn generalized eig problem into eigh (the L_inv stuff) is similar to the initial guess stuff. Unfortunately, this has the downside of removing variable names so less documentation.